File: 프로그래머스/lv2/131704. 택배상자/택배상자.java
import java.util.*;
class Solution {
    public int solution(int[] order) {
        int answer = 0;
        int now =0;
        Stack<Integer> container = new Stack<>();
        for(int i=1;i<=order.length;i++){
            if(i!=order[now]){
                container.push(i);
            }else if(i==order[now]){
                answer+=1;
                now +=1;
            }
            if(!container.isEmpty()&& container.peek() == order[now]){
                    container.pop();
                    answer+=1;
                    now+=1;
            }
        }
        while(true){
            if(!container.isEmpty()&&container.peek()==order[now]){
                container.pop();
                answer+=1;
                now+=1;
            }
            else{
                break;
            }
        }
        
        
        return answer;
    }
}

File: 프로그래머스/lv2/131704. 택배상자/택배상자.py
def solution(order):
    answer = 0
    stack = []

    now = 0
    for i in range(1,len(order)+1):
        if(i != order[now]):
            stack.append(i)
        elif(i == order[now]) :
            answer+=1
            now+=1;

        if(len(stack)>0 and stack[-1]==order[now]):
            answer+=1
            stack.pop()
            now+=1;
    
    while True:
        if(len(stack)>0 and stack[-1] ==order[now]):
            answer+=1
            stack.pop()
            now+=1
        else:
            break;

    return answer